lib/active_model/serializer/reflection.rb

```diff
@@ -106,17 +106,19 @@ def value(serializer)
       #
       # @api private
       #
-      def build_association(subject, parent_serializer_options)
-        association_value = value(subject)
+      def build_association(parent_serializer, parent_serializer_options)
+        association_value = value(parent_serializer)
         reflection_options = options.dup
-        serializer_class = subject.class.serializer_for(association_value, reflection_options)
+        serializer_class = parent_serializer.class.serializer_for(association_value, reflection_options)
```

Comment:
I sort of don't think it should a method on the serializer class in the first place.. but I've also seen people override the method per serializer.  dunno